shared/desktop/webpack.config.babel.js

```diff
@@ -160,24 +169,46 @@ const config = (_, {mode}) => {
     },
     target: 'electron-main',
   })
-  const renderThreadConfig = merge(commonConfig, {
-    context: path.resolve(__dirname, '..'),
-    devtool: isDev ? 'eval' : 'source-map',
-    entry: {
-      'component-loader': './desktop/remote/component-loader.desktop.js',
-      index: './desktop/renderer/index.desktop.js',
-    },
+
+  const hmrPlugin = isHot && isDev ? [new webpack.HotModuleReplacementPlugin()] : []
+  const template = path.join(__dirname, './renderer/index.html.template')
+  const makeHtmlName = name => `${name}${isDev ? '.dev' : ''}.html`
+  const makeViewPlugins = names =>
+    [
+      ...hmrPlugin,
+      // Map since we generate multiple html files
+      ...names.map(
+        name =>
+          new HtmlWebpackPlugin({
+            filename: makeHtmlName(name),
+            name,
+            inject: false,
+            isDev,
+            template,
+          })
+      ),
+    ].filter(Boolean)
+
+  // just keeping main in its old place
+  const entryOverride = {
+    main: 'desktop/renderer',
+  }
+
+  // multiple entries so we can chunk shared parts
+  const entries = ['main', 'tracker', 'menubar', 'pinentry', 'unlock-folders']
+  const viewConfig = merge(commonConfig, {
+    entry: entries.reduce((map, name) => {
+      map[name] = `./${entryOverride[name] || name}/main.desktop.js`
+      return map
+    }, {}),
+    optimization: {splitChunks: {chunks: 'all'}},
```

Comment:
we now generate shared chunks so we don't have multiple copies of code between all our js bundles